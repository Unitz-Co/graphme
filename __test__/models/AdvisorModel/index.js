import _ from 'lodash';
import graphme from '../../../index';

import Definition from '../Definition';

import GetAdvisor from '../../gql/advisor/GetAdvisor.gql';

import AdvisorProfileModel from '../AdvisorProfileModel';
import AdvisorPresenceModel from '../AdvisorPresenceModel';
import TransactionModel from '../TransactionModel';
import TransactionAggregateModel from '../TransactionAggregateModel';

class AdvisorModel extends graphme.BaseModel {
  static DEFINITION = Definition.create({
    name: 'Advisor',
    schema: {
      id: String,
      is_active: Boolean,
      created_at: Date,
    },
    nodes: [
      ['profile', 'AdvisorProfileModel', { column_mapping: { id: 'advisor_id' } }],
      ['presence', 'AdvisorPresenceModel', { column_mapping: { id: 'advisor_id' } }],
      ['transactions', 'TransactionModel.Collection', { column_mapping: { id: 'advisor_id' } }],
      [
        'transactions_aggregate',
        'TransactionAggregateModel',
        { column_mapping: { id: 'advisor_id' }, usePlanSync: true },
      ],
    ],
    key: 'id',

    baseQuery: GetAdvisor,

    GQL_ACTIONS: {
      FIND: `advisor`,
      GET: `advisor_by_pk`,
      INSERT: `insert_advisor_one`,
      UPDATE: `update_advisor_by_pk`,
      DELETE: `delete_advisor_by_pk`,
    },
  });
}

graphme.model('AdvisorModel', AdvisorModel);

export default AdvisorModel;
