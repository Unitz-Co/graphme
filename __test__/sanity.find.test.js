import AdvisorModel from './models/AdvisorModel';

test('init with find gql string', async () => {
  const advisors = await AdvisorModel.find(`limit: 1, where: { id: { _eq: "PdOJWFBgNPUEMhX1JlsDm7zWy012" } } `);

  const advisor = await advisors.getByPath('0');
  console.snapshot('advisor.0.profile.display_name', await advisor.getByPath('profile.display_name'));
  console.snapshot('advisor list', advisors);
  console.snapshot('advs', await advisors.getByPath('0.profile.display_name'));
  const transactions = await advisors.getByPath('0.transactions');

  await transactions.setArgs('limit: 2').sync();
  console.snapshot('transactions(limit:2).length', transactions.length);

  await transactions.setArgs('limit: 6').sync();
  console.snapshot('transactions(limit:6).length', transactions.length);
});

test('init with find gql object', async () => {
  const advisors = await AdvisorModel.find(
    { limit: 5, where: { id: { _eq: 'PdOJWFBgNPUEMhX1JlsDm7zWy012' } } },
    ({ node }) => node.merge(`transactions { id }`)
  );

  const advisor = await advisors.getByPath('0');
  console.snapshot('advisor.0.profile.display_name', await advisor.getByPath('profile.display_name'));
  console.snapshot('advisor list', advisors);
  console.snapshot('advs', await advisors.getByPath('0.profile.display_name'));
  const transactions = await advisors.getByPath('0.transactions');

  await transactions.setArgs('limit: 2').sync();
  console.snapshot('transactions.length', transactions.length);
});
