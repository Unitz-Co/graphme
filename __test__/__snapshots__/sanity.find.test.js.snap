// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init with find gql object 1`] = `
Object {
  "args": Array [
    "advisor.0.profile.display_name",
    "new_displayName",
  ],
}
`;

exports[`init with find gql object 2`] = `
Object {
  "args": Array [
    "advisor list",
    CollectionMixins [
      AdvisorModel {},
    ],
  ],
}
`;

exports[`init with find gql object 3`] = `
Object {
  "args": Array [
    "advs",
    "new_displayName",
  ],
}
`;

exports[`init with find gql object 4`] = `
Object {
  "args": Array [
    "transactions.length",
    2,
  ],
}
`;

exports[`init with find gql string 1`] = `
Object {
  "args": Array [
    "advisor.0.profile.display_name",
    "new_displayName",
  ],
}
`;

exports[`init with find gql string 2`] = `
Object {
  "args": Array [
    "advisor list",
    CollectionMixins [
      AdvisorModel {},
    ],
  ],
}
`;

exports[`init with find gql string 3`] = `
Object {
  "args": Array [
    "advs",
    "new_displayName",
  ],
}
`;

exports[`init with find gql string 4`] = `
Object {
  "args": Array [
    "transactions(limit:2).length",
    2,
  ],
}
`;

exports[`init with find gql string 5`] = `
Object {
  "args": Array [
    "transactions(limit:6).length",
    6,
  ],
}
`;

exports[`list advisors 1`] = `
Object {
  "args": Array [
    "list advisors",
    CollectionMixins [
      AdvisorModel {},
      AdvisorModel {},
      AdvisorModel {},
      AdvisorModel {},
      AdvisorModel {},
      AdvisorModel {},
    ],
  ],
}
`;
